src/plugins/info/info_impl.cpp

```diff
@@ -172,12 +172,13 @@ std::string InfoImpl::swap_and_translate_binary_to_str(uint8_t* binary, unsigned
 
 std::string InfoImpl::translate_binary_to_str(uint8_t* binary, unsigned binary_len)
 {
-    std::string str(binary_len * 2, '0');
+    std::string str(binary_len * 2 + 1, '0');
 
     for (unsigned i = 0; i < binary_len; ++i) {
         // One hex number occupies 2 chars.
         snprintf(&str[i * 2], str.length() - i * 2, "%02x", binary[i]);
     }
+    snprintf(&str[binary_len * 2], str.length() - binary_len * 2 - 1, "%x", '\0');
```

Comment:
```suggestion
```

I'm pretty sure it's superfluous to add `\0` at the end of an `std::string` (which is what we return here). As mentioned, `\0` is not part of an `std::string`, and it is just added when doing the `c_str()` conversion.